### PR TITLE
Indexes file information from PDC Describe records

### LINKED ISSUE #378: Import files information from PDC Describe record
We now get the file information from PDC Describe and we should be able to index it (see https://github.com/pulibrary/pdc_discovery/blob/main/lib/traject/pdc_describe_indexing_config.rb#L356-L371 )

### PR DESCRIPTION
**Pull request by hectorcorrea**
Indexes and displays file information for records coming from PDC Describe:

show page with link to files

Closes #378

### CODE REVIEW

File: lib/traject/pdc_describe_indexing_config.rb [hunk 1]
```diff
@@ -342,38 +342,22 @@
...
-#   bitstreams.reject! do |bitstream|
-#     bitstream[:bundle_name] != "ORIGINAL" || bitstream[:name] == "license.txt"
-#   end
-
-#   accumulator.concat [bitstreams.to_json.to_s]
-# end
+# ==================
+# Store all files metadata as a single JSON string so that we can display detailed information for each of them.
+to_field 'files_ss' do |record, accumulator, _context|
+  # TODO: Remove prefix once we export the complete URI in PDC Describe
```
Comment (hectorcorrea) [inline lib/traject/pdc_describe_indexing_config.rb:348]: This issue will take care of the TODO https://github.com/pulibrary/pdc_describe/issues/902

File: spec/lib/describe_indexer_spec.rb [hunk 2]
```diff
@@ -95,12 +95,24 @@
...
+        files = JSON.parse(indexed_record['files_ss'])
+        file1 = files.find { |file| file["name"] == "file1.jpg" }
+        file2 = files.find { |file| file["name"] == "file2.txt" }
+        expect(file1["size"]).to eq "316781"
+        expect(file2["size"]).to eq "396003"
+      end
+    end
+
+    context "all text catch all field" do
+      it "indexes the all text catch all field" do
```
Comment (carolyncole) [inline spec/lib/describe_indexer_spec.rb:108]:
```suggestion
      it "indexes the file name in the all text catch all field" do
```

File: spec/system/dc_fields_indexing_spec.rb [hunk 3]
```diff
@@ -3,6 +3,8 @@
 # rubocop:disable Metrics/BlockLength
 describe 'DataSpace research data all fields indexing', type: :system do
   subject(:result) do
+    Blacklight.default_index.connection.delete_by_query("*:*")
```
Comment (carolyncole) [inline spec/system/dc_fields_indexing_spec.rb:6]: Just curious why this is needed now...

File: lib/traject/pdc_describe_indexing_config.rb (hunk 1 above)
Comment (carolyncole) [reply lib/traject/pdc_describe_indexing_config.rb:348]: Is there a ticket for updating the todo, or should we wait on this PR?

**Review by carolyncole — CHANGES_REQUESTED** [review]
Looking great @hectorcorrea!

One suggested change in the test, and a couple of questions.

File: lib/traject/pdc_describe_indexing_config.rb (hunk 1 above)
Comment (hectorcorrea) [reply lib/traject/pdc_describe_indexing_config.rb:348]: I am planning on making the change as I work on 902.

**Review by carolyncole — APPROVED** [review]

**Pull request merged**